### DISCUSSION TITLE
Change Detection LXC: broken script ?

### DISCUSSION THREAD

**Discussion by miketrump** [tteck/Proxmox#3772] [Help Needed]
Hello
I just use Change Detection LXC script (bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/changedetection.sh)")
But I got an error mid install, here is my last output:

` ✓ Installed Change Detection
 /Collecting playwrightss & Playwright   
 \  Downloading playwright-1.47.0-py3-none-manylinux1_x86_64.whl (38.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 38.1/38.1 MB 5.9 MB/s eta 0:00:00
 |Collecting greenlet==3.0.3
  Downloading greenlet-3.0.3-cp311-cp311-manylinux_2_24_x86_64.manylinux_2_28_x86_64.whl (620 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 620.0/620.0 kB 6.9 MB/s eta 0:00:00
 \Collecting pyee==12.0.0
  Using cached pyee-12.0.0-py3-none-any.whl (14 kB)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.11/dist-packages (from pyee==12.0.0->playwright) (3.10.0.2)
 \Installing collected packages: pyee, greenlet, playwright
  Attempting uninstall: pyee
    Found existing installation: pyee 11.1.1
    Uninstalling pyee-11.1.1:
      Successfully uninstalled pyee-11.1.1
  Attempting uninstall: greenlet
    Found existing installation: greenlet 3.1.1
 |    Uninstalling greenlet-3.1.1:
      Successfully uninstalled greenlet-3.1.1
 |ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pyppeteer-ng 2.0.0rc5 requires pyee<12.0.0,>=11.0.0, but you have pyee 12.0.0 which is incompatible.
Successfully installed greenlet-3.0.3 playwright-1.47.0 pyee-12.0.0
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
 /Cloning into '/opt/browserless'...
 |remote: Enumerating objects: 23457, done.
remote: Counting objects: 100% (4396/4396), done.
remote: Compressing objects: 100% (418/418), done.
 -error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
error: 5186 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output

[ERROR] in line 77: exit code 0: while executing command $STD git clone https://github.com/browserless/chrome /opt/browserless`

Is it a bug to be fixed? or I have to fix something?
I did rerun the script and get the same error
Thanks

**Comment by lbeier** [comment]
I have a similar issues in the `update` command.

```
   ________                              ____       __            __  _           
  / ____/ /_  ____ _____  ____ ____     / __ \___  / /____  _____/ /_(_)___  ____ 
 / /   / __ \/ __ `/ __ \/ __ `/ _ \   / / / / _ \/ __/ _ \/ ___/ __/ / __ \/ __ \
/ /___/ / / / /_/ / / / / /_/ /  __/  / /_/ /  __/ /_/  __/ /__/ /_/ / /_/ / / / /
\____/_/ /_/\__,_/_/ /_/\__, /\___/  /_____/\___/\__/\___/\___/\__/_/\____/_/ /_/ 
                       /____/                                                     
 - Updating Change Detection LXC   
[ERROR] in line 66: exit code 0: while executing command git -C /opt/browserless/ pull &> /dev/null

root@changedetection:~# 
```

**Reply by tteck** [reply]
🏁 The script runs without any problems from this end.

```
   ________                              ____       __            __  _           
  / ____/ /_  ____ _____  ____ ____     / __ \___  / /____  _____/ /_(_)___  ____ 
 / /   / __ \/ __ `/ __ \/ __ `/ _ \   / / / / _ \/ __/ _ \/ ___/ __/ / __ \/ __ \
/ /___/ / / / /_/ / / / / /_/ /  __/  / /_/ /  __/ /_/  __/ /__/ /_/ / /_/ / / / /
\____/_/ /_/\__,_/_/ /_/\__, /\___/  /_____/\___/\__/\___/\___/\__/_/\____/_/ /_/ 
                       /____/                                                     
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 111
Using Hostname: changedetection
Using Disk Size: 8
Allocated Cores: 2
Allocated RAM: 1024
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Change Detection LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 111 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.114.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Updated Python3
 ✓ Set up Node.js Repository
 ✓ Installed Node.js
 ✓ Installed Change Detection
 ✓ Installed Browserless & Playwright
 ✓ Installed Font Packages
 ✓ Installed X11 Packages
 ✓ Created Services
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!

Change Detection should be reachable by going to the following URL.
         http://192.168.86.250:5000 
```
```
   ________                              ____       __            __  _           
  / ____/ /_  ____ _____  ____ ____     / __ \___  / /____  _____/ /_(_)___  ____ 
 / /   / __ \/ __ `/ __ \/ __ `/ _ \   / / / / _ \/ __/ _ \/ ___/ __/ / __ \/ __ \
/ /___/ / / / /_/ / / / / /_/ /  __/  / /_/ /  __/ /_/  __/ /__/ /_/ / /_/ / / / /
\____/_/ /_/\__,_/_/ /_/\__, /\___/  /_____/\___/\__/\___/\___/\__/_/\____/_/ /_/ 
                       /____/                                                     
 ✓ Updated Successfully
root@changedetection:~# 
```

**Reply by birukoff** [reply]
I can confirm, Change Detection script in **UPDATE** mode is broken. There is no way to enable verbose logging in **UPDATE** mode. There is no advanced settings. How do you enable it?
The message: `[ERROR] in line 69: exit code 0: while executing command npm run build:function --prefix /opt/browserless &> /dev/null`

**Reply by lbeier** [reply]
Removing the LXC and reinstalling it worked, and the update command works fine now.

**Reply by birukoff** [reply]
Reinstalling means configuration is lost.

**Reply by lbeier** [reply]
> Reinstalling means configuration is lost.

That’s true. I didn’t have any configuration, so it was a no brainier for me. The tool has a backup option, but I’m not sure how to restore it. It’s probably out of scope of this issue though.
